src/libEth/Eth.cpp

```diff
@@ -145,6 +159,178 @@ EthFields parseRawTxFields(std::string const &message) {
   return ret;
 }
 
+EthFields parseEip2930Transaction(zbytes asBytes) {
+  EthFields ret;
+
+  dev::RLP rlpStream1(asBytes,
+                      dev::RLP::FailIfTooBig | dev::RLP::FailIfTooSmall);
+
+  if (rlpStream1.isNull()) {
+    LOG_GENERAL(WARNING, "Failed to parse RLP stream in raw TX!");
+    return {};
+  }
+
+  int i = 0;
+
+  ret.version = DataConversion::Pack(CHAIN_ID, TRANSACTION_VERSION_ETH_EIP_2930);
+
+  // RLP TX contains: chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, signatureYParity, signatureR, signatureS
+  for (auto it = rlpStream1.begin(); it != rlpStream1.end();) {
+    auto byteIt = (*it).operator zbytes();
```

Comment:
Why is this here? looks like it's only used in case 6; You can just use:
`ret.code = static_cast<...>(*it)` there which will invoke the operator.

In addition, why not just have: `for (const auto& value = rlpStream1) { ... }`? You don't seem to be manipulating the iterator other than incrementing it.

Reply:
Copy pasted from the other parsing code. I will simplify :+1: